core/src/Kokkos_SYCL_Space.hpp

```diff
@@ -82,6 +83,9 @@ class SYCLDeviceUSMSpace {
  public:
   static constexpr const char* name() { return "SYCLDeviceUSM"; };
 
+  static void access_error();
```

Comment:
can we name these impl_... This is not part of the general public interface or?

Reply:
See #3813 where we try to remove `VerifyExecutionCanAccessMemorySpace` and `access_error` is the only place that uses `access_error` for `HIP` and `CUDA` internally. We also want to deprecate that function there and if we decide to do that, I would just remove it here.

Reply:
It only exists in cuda and hip otherwise anyway so no need to introduce it here.